texture.h

```diff
@@ -6,13 +6,20 @@ class Texture
 {
 private:
     std::string mFilePath;
-    unsigned char* mLocalBuffer;
-    int mWidth, mHeight, mBPP;
+    unsigned char* mLocalBuffer { nullptr };
 protected:
-    unsigned int mRendererID;
+    int mWidth {0};
+    int mHeight {0};
+    int mBPP {0};
+    unsigned int mRendererID {0};
 public:
+    Texture();
```

Comment:
I think here you can possibly say 

```
Texture() = default;
```

To get rid of the empty one in the source file. Not like it's a massive difference or anything